internal/auth0/user/resource.go

```diff
@@ -470,7 +473,7 @@ func validateNoPasswordAndEmailVerifiedSimultaneously() validateUserFunc {
 	}
 }
 
-func updateUserRoles(d *schema.ResourceData, api *management.Management) error {
+func persistUserRoles(d *schema.ResourceData, api *management.Management) error {
```

Comment:
This was conflict with updateUserRoles func for the user_roles resource, so I just renamed it. However this logic will likely go inside the auth0_user_roles file after it is removed from here.